src/lib/sql/sql_pipeline.hpp

```diff
@@ -14,6 +14,16 @@
 
 namespace opossum {
 
+// Holds relevant information about the execution of an SQLPipeline.
+struct SQLPipelineMetrics {
+  std::vector<std::reference_wrapper<const SQLPipelineStatementMetrics>> statement_metrics;
```

Comment:
Any reason why you are not using smart ptrs or just plain instances here? references are dangerous!

Reply:
plain instances would be copies and then changes in the original statements are not reflected here. i guess I could do `shared_ptr`s but they kinda really aren't.  al long as the pipeline exists, the statements exist and thus the references exist. but if you have a strong feeling towards shared_ptr, I'll change it

Reply:
It's data shared between the `SQLPipelineMetrics` and the `SQLPipelineStatement` - `shared_ptr` it is. That way no one needs to know about the implicit connections between SQLPipelineMetrics and Pipelines and PipelineStatements and no one can (this is important) break them while changing this code.